### PR TITLE
fix(ai proxy): 3.7 regression fixes rollup

### PR DESCRIPTION
**Pull request by tysoekong**
### Summary

Regression test fix rollup from Kong 3.7 Release Candidate 1.

### Checklist

- [x] The Pull Request has tests

### Issue reference

- KAG-4356
- FTI-5767
- FTI-5769

### CODE REVIEW

File: kong/llm/drivers/anthropic.lua [hunk 1]
```diff
@@ -209,8 +209,7 @@ local function handle_stream_event(event_t, model_info, route_type)
     and event_data.usage then
       return nil, nil, {
         prompt_tokens = nil,
-        completion_tokens = event_data.meta.usage
-                        and event_data.meta.usage.output_tokens
+        completion_tokens = event_data.usage.output_tokens
```
Comment (jschmid1) [inline kong/llm/drivers/anthropic.lua:212]: This seems more than just syntax correction, eh? Can you give a hint why you did this?

**Review by jschmid1 — APPROVED** [review]
Looks good, if you could address my inline comment?

File: kong/llm/drivers/anthropic.lua (hunk 1 above)
Comment (tysoekong) [reply kong/llm/drivers/anthropic.lua:212]: @jschmid1 

It is a mistake :(

One of the earlier "event types" has data.meta.usage, whereas this one has data.usage, for its token reporting.

Not entirely sure why they decided to use different messaging formats for similar things, but when I ctrl+cv'd it I didn't notice they were different.

Comment (jschmid1) [reply kong/llm/drivers/anthropic.lua:212]: Sounds like we'd benefit from an additional test here :)

Comment (tysoekong) [reply kong/llm/drivers/anthropic.lua:212]: It will be covered in 3.8 real integration tests, right now we have Insomnia suite running through everything.

**Review by hanshuebner — APPROVED** [review]

**Pull request merged**